builtin/aws/ecs/platform.go

```diff
@@ -773,9 +773,14 @@ func (p *Platform) Launch(
 	}
 
 	s.Update("Registered Task definition: %s", family)
-	rand := id[len(id)-(31-len(app.App)):]
 
-	serviceName := fmt.Sprintf("%s-%s", app.App, rand)
+	serviceName := fmt.Sprintf("%s-%s", app.App, id)
+
+	// We have to clamp at a length of 32 because the Name field to CreateTargetGroup
+	// requires that the name is 32 characters or less.
+	if len(serviceName) > 32 {
+		serviceName = serviceName[:32]
```

Comment:
Maybe worth logging that the name is getting trimmed here due to a limitation in `CreateTargetGroup`?

```suggestion
                L.Info("Trimming service name to 32 characters due to limitation in 'CreateTargetGroup' name field")
		serviceName = serviceName[:32]
```

Reply:
Nah, because the user doesn't really need to even know about this value as it's just fed to the targetgroup name.

Reply:
Ahhh, cool :+1: Yeah, if they don't know about this value they probably won't care that it got trimmed